src/components/checkbox/checkbox.tsx

```diff
@@ -73,7 +73,7 @@ export class Checkbox {
         this.formField.input = this.mdcCheckbox;
     }
 
-    public componentDidUnload() {
+    public disconnectedCallback() {
         this.mdcCheckbox.destroy();
         this.formField.destroy();
     }
```

Comment:
Hmm… with this fix, I think we'll have to run the setup on `connectedCallback` too, right?

Reply:
Hmm... maybe :thinking:  Good catch either way! :slightly_smiling_face: 
I guess it depends on if `disconnectedCallback` behaves any different than `componentDidUnload` :thinking: 
The reason for this change was that the Stencil ruleset complained about `componentDidUnload` being public, and after some research I found out that `componentDidUnload` will be deprecated in the future. It's not event documented in the Stencil documentation any longer.

However, `connectedCallback` will be called whenever the element is inserted into the DOM while `componendWillLoad` is only called once. So yeah on a second thought I guess that means we should update all of them if we use `disconnectedCallback` like this :smile:

Reply:
> after some research I found out that `componentDidUnload` will be deprecated in the future. It's not event documented in the Stencil documentation any longer.


I thought `componentDidUnload` didn't even exist, but this explains why I once started using it 😄

> However, `connectedCallback` will be called whenever the element is inserted into the DOM while `componendWillLoad` is only called once. So yeah on a second thought I guess that means we should update all of them if we use `disconnectedCallback` like this 😄

👍